Author: Sune Vuorela <sune@debian.org>
Description: use -gstabs debugging symbols on s390
 Build s390 debugging symbols with -gstabs instead of with -g to reduce linker
 memory usage. Thanks to Oswald Buddenhagen for the hint.
Bug-Debian: http://bugs.debian.org/528485
Origin: vendor
Forwarded: no

--- a/src/3rdparty/webkit/WebKit.pri
+++ b/src/3rdparty/webkit/WebKit.pri
@@ -47,6 +47,13 @@ CONFIG(release, debug|release) {
     DEFINES += NDEBUG
 }
 
+equals(QT_ARCH, s390) {
+	message("S/390 workaround for QtWebkit: do not build with -g, but with -gstabs")
+	QMAKE_CXXFLAGS -= -g
+	QMAKE_CXXFLAGS += -gstabs
+}
+
+
 BASE_DIR = $$PWD
 INCLUDEPATH += $$OUTPUT_DIR/include/QtWebKit
 QMAKE_LFLAGS += -Wl,--version-script,$$BASE_DIR/symbols.filter
